core/base-service/auth-helper.js

```diff
@@ -1,14 +1,21 @@
 'use strict'
 
 class AuthHelper {
-  constructor({ userKey, passKey, isRequired = false }, privateConfig) {
+  constructor(
+    { userKey, passKey, isRequired = false, useEmptyStringForUser = false },
```

Comment:
I find the variable name `useEmptyStringForUser` slightly confusing, in the sense that it will only use empty string if the `userKey` is not specified, else it won't use an empty string. Would `defaultToEmptyStringForUser` or something like that better convey the behaviour?